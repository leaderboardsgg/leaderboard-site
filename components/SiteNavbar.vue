<template>
  <div class="border-b border-gray-300">
    <div
      class="
        container relative
        flex flex-wrap lg:flex-row justify-between
        mx-auto px-3 py-3
      "
    >
      <NuxtLink class="flex flex-grow" to="/">
        <div class="flex">
          <svg
            xmlns="http://www.w3.org/2000/svg"
            class="h-10 w-10 mx-1"
            fill="none"
            viewBox="0 0 24 24"
            stroke="currentColor"
          >
            <path
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"
            />
          </svg>
          <h1
            class="
              w-full
              py-2
              text-lg text-center md:text-left
              font-semibold
              leading-6
              tracking-tight
              hidden md:inline
            "
          >
            speedrun.website
          </h1>
        </div>
      </NuxtLink>
      <div
        v-show="lg || mobileNavIsActive"
        class="
          border-t border-gray-300 lg:border-t-0
          flex flex-col lg:flex-row
          mt-2 lg:mt-0
          w-full lg:w-auto
          space-y-2 lg:space-y-0
        "
      >
        <CoreNavLinks />
        <CoreSearchBar />
      </div>
      <div class="
             flex flex-row items-center
             absolute right-2 top-4
             lg:relative lg:right-0 lg:top-0
             space-x-2 lg:space-x-0
           "
      >
        <CoreLoginButton />
        <CoreSignUpButton />
        <button
          class="
            items-center
            rounded
            text-black
            lg:hidden
          "
          @click="mobileNavIsActive = !mobileNavIsActive"
        >
          <svg
            class="fill-current h-5 w-5 mx-2"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>Menu</title>
            <path d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0v-2z" />
          </svg>
        </button>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'

/**
 * TODO use https://vueuse.org/core/usebreakpoints and
 * https://vueuse.org/core/useEventListener/
 */

export default Vue.extend({
  data: () => ({
    lg: false,
    mobileNavIsActive: false,
  }),
  mounted() {
    this.lgSetBool()
  },
  beforeMount() {
    addEventListener('resize', this.lgSetBool)
  },
  beforeDestroy() {
    removeEventListener('resize', this.lgSetBool)
  },
  methods: {
    lgSetBool() {
      return (this.lg = innerWidth >= 1024)
    },
  },
})
</script>
