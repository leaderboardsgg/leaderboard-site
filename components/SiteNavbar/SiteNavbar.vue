<template>
  <div class="border-b border-gray-300">
    <div
      class="
        container
        relative
        flex flex-wrap
        lg:flex-row
        justify-between
        mx-auto
        px-3
        py-3
      "
    >
      <div class="flex flex-grow">
        <NuxtLink class="flex" to="/">
          <svg
            xmlns="http://www.w3.org/2000/svg"
            class="h-10 w-10 mx-1"
            fill="none"
            viewBox="0 0 24 24"
            stroke="currentColor"
          >
            <path
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"
            />
          </svg>
          <h1
            class="
              w-full
              py-2
              text-lg text-center
              md:text-left
              font-semibold
              leading-6
              tracking-tight
              hidden
              md:inline
            "
          >
            leaderboards.gg
          </h1>
        </NuxtLink>
      </div>
      <div
        :class="{ hidden: !mobileNavIsActive }"
        class="
          lg:flex
          border-t border-gray-300
          lg:border-t-0
          flex-col
          lg:flex-row
          mt-2
          lg:mt-0
          w-full
          lg:w-auto
          space-y-2
          lg:space-y-0
        "
      >
        <CoreNavLinks />
        <CoreSearchBar />
      </div>
      <div
        class="
          flex flex-row
          items-center
          absolute
          right-2
          top-4
          lg:relative lg:right-0 lg:top-0
          space-x-2
          lg:space-x-0
        "
      >
        <CoreLoginButton
          data-testId="site-navbar-login-button"
          @click="toggleLoginModal"
          @keyup.enter="toggleLoginModal"
        />
        <CoreSignUpButton
          data-testId="site-navbar-sign-up-button"
          @click="toggleSignUpModal"
          @keyup.enter="toggleSignUpModal"
        />

        <button
          class="items-center rounded text-black lg:hidden"
          @click="toggleMenu"
        >
          <svg
            class="fill-current h-5 w-5 mx-2"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>Menu</title>
            <path d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0v-2z" />
          </svg>
        </button>
      </div>
    </div>

    <transition
      enter-active-class="transition-opacity duration-200"
      leave-active-class="transition-opacity duration-200"
      enter-class="opacity-0"
      enter-to-class="opacity-100"
      leave-class="opacity-100"
      leave-to-class="opacity-0"
    >
      <BaseModal v-show="showModal" @close="showModal = false">
        <LogInCard
          v-show="showLogin"
          class="shadow-xl"
          :modal="true"
          @close="showModal = false"
          @signUpClick="showLogin = false"
        />
        <SignUpCard
          v-show="!showLogin"
          class="shadow-xl"
          :modal="true"
          @close="showModal = false"
          @logInClick="showLogin = true"
        />
      </BaseModal>
    </transition>
  </div>
</template>

<script lang="ts">
import { defineComponent, reactive, toRefs } from '@nuxtjs/composition-api';
import LogInCard from '../LogInCard/LogInCard.vue';
import SignUpCard from '../SignUpCard/SignUpCard.vue';

export default defineComponent({
  components: {
    LogInCard,
    SignUpCard,
  },
  setup() {
    const state = reactive({
      mobileNavIsActive: false,
      showLogin: false,
      showModal: false,
    });

    function toggleMenu() {
      state.mobileNavIsActive = !state.mobileNavIsActive;
    }

    function toggleLoginModal() {
      state.showModal = !state.showModal;
      state.showLogin = true;
    }

    function toggleSignUpModal() {
      state.showModal = !state.showModal;
      state.showLogin = false;
    }

    return {
      ...toRefs(state),
      toggleLoginModal,
      toggleMenu,
      toggleSignUpModal,
    };
  },
});
</script>
