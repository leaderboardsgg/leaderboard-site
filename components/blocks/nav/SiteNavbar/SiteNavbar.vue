<script setup lang="ts">
import { computed, reactive } from 'vue'
import NavLinks from 'elements/nav/NavLinks/NavLinks.vue'
import LogoutButton from 'elements/buttons/LogoutButton/LogoutButton.vue'
import SignUpButton from 'elements/buttons/SignUpButton/SignUpButton.vue'
import LoginButton from 'elements/buttons/LoginButton/LoginButton.vue'
import LogInCard from 'blocks/cards/LogInCard/LogInCard.vue'
import SignUpCard from 'blocks/cards/SignUpCard/SignUpCard.vue'
import BaseModal from 'elements/modals/BaseModal/BaseModal.vue'
import SearchBar from 'blocks/SearchBar/SearchBar.vue'
import { useLogoutUser } from 'composables/api'
import { User } from 'lib/api/data-contracts'

interface NavbarState {
  mobileNavIsActive: boolean
  showLogin: boolean
  showModal: boolean
}

const state: NavbarState = reactive({
  mobileNavIsActive: false,
  showLogin: false,
  showModal: false,
})

const currentUser = useState<User>('current_user', () => ({
  admin: false,
  email: '',
  username: '',
}))
const loggedIn = computed(
  () => !!currentUser.value?.username && currentUser.value?.username !== '',
)

function toggleMenu() {
  state.mobileNavIsActive = !state.mobileNavIsActive
}

function toggleLoginModal() {
  state.showModal = !state.showModal
  state.showLogin = true
}

function toggleSignUpModal() {
  state.showModal = !state.showModal
  state.showLogin = false
}

function logout() {
  useLogoutUser()
}
</script>

<template>
  <div class="site-navbar">
    <div class="site-navbar__wrapper">
      <div class="site-logo">
        <NuxtLink class="site-navbar__homelink" to="/">
          <i-svg-clock class="mx-1 h-10 w-10" />
          <h1>leaderboards.gg</h1>
        </NuxtLink>
      </div>
      <div :class="{ hidden: !state.mobileNavIsActive }" class="mobile-menu">
        <NavLinks />
        <SearchBar />
      </div>
      <div class="mobile-navbar">
        <LoginButton
          v-if="!loggedIn"
          data-testId="site-navbar-login-button"
          @click="toggleLoginModal"
          @keyup.enter="toggleLoginModal"
        />
        <SignUpButton
          v-if="!loggedIn"
          data-testId="site-navbar-sign-up-button"
          @click="toggleSignUpModal"
          @keyup.enter="toggleSignUpModal"
        />
        <LogoutButton
          v-if="loggedIn"
          data-testId="site-navbar-logout-button"
          @click="logout"
          @keyup.enter="logout"
        />

        <button class="mobile-menu__toggle" @click="toggleMenu">
          <i-svg-menu class="mx-2 h-5 w-5 fill-current" />
        </button>
      </div>
    </div>

    <transition
      v-if="!loggedIn"
      enter-active-class="transition-opacity duration-200"
      leave-active-class="transition-opacity duration-200"
      enter-class="opacity-0"
      enter-to-class="opacity-100"
      leave-class="opacity-100"
      leave-to-class="opacity-0"
    >
      <BaseModal v-show="state.showModal" @close="state.showModal = false">
        <LogInCard
          v-show="state.showLogin"
          class="shadow-xl"
          :modal="true"
          @close="state.showModal = false"
          @sign-up-click="state.showLogin = false"
        />
        <SignUpCard
          v-show="!state.showLogin"
          class="shadow-xl"
          :modal="true"
          @close="state.showModal = false"
          @log-in-click="state.showLogin = true"
        />
      </BaseModal>
    </transition>
  </div>
</template>

<style lang="postcss" scoped>
.site-navbar {
  @apply border-b border-gray-300;
  & .site-navbar__wrapper {
    @apply container relative flex flex-wrap lg:flex-row justify-between mx-auto px-3 py-3;
    & .mobile-menu {
      @apply lg:flex border-t border-gray-300 lg:border-t-0 flex-col lg:flex-row mt-2 lg:mt-0 w-full lg:w-auto space-y-2 lg:space-y-0;
    }
    & .mobile-navbar {
      @apply flex flex-row items-center absolute right-2 top-4 lg:relative lg:right-0 lg:top-0 space-x-2 lg:space-x-0;
      & .mobile-menu__toggle {
        @apply items-center rounded text-black lg:hidden;
      }
    }
  }
  & .site-logo {
    @apply flex grow;
    & .site-navbar__homelink {
      @apply flex;
      & h1 {
        @apply w-full py-2 text-lg text-center md:text-left font-semibold leading-6 tracking-tight hidden md:inline;
      }
    }
  }
}
</style>
