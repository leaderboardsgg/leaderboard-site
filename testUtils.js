import Vue from 'vue';
import { RouterLinkStub } from '@vue/test-utils';
import userEvent from '@testing-library/user-event';
import { render } from '@testing-library/vue';

export const NuxtLinkStub = Vue.component('NuxtLinkStub', {
  extends: RouterLinkStub,
  render(h) {
    const temp = h(
      this.tag || 'a',
      {
        attrs: {
          ...this.$vnode.data.attrs,
          ...this.$attrs,
          href: this.to,
        },
      },
      this.$slots.default,
    );

    return temp;
  },
});

export const stubbedRender = (Component, options) => {
  return render(Component, {
    stubs: { NuxtLink: NuxtLinkStub, ...options?.stubs },
    ...options,
  });
};

export * from '@testing-library/vue';
export * from '@vue/test-utils';
export { userEvent as fireEvent };
