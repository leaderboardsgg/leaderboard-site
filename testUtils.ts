import { createElementBlock, Component, ComponentOptions } from 'vue'
import { MountingOptions } from '@vue/test-utils'
import userEvent from '@testing-library/user-event'
import { render, RenderOptions } from '@testing-library/vue'

export const stubbedRender = (
  Component: Component | ComponentOptions,
  options?: RenderOptions | MountingOptions<any>,
) => {
  return render(Component, {
    global: {
      stubs: {
        NuxtLink: createElementBlock('a'),
      },
    },
    ...options,
  })
}

export * from '@testing-library/vue'
export * from '@nuxt/test-utils'
export * from '@vue/test-utils'
export { userEvent as fireEvent }
