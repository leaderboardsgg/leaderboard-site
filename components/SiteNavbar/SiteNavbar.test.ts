import { stubbedRender } from '@/testUtils';
import SiteNavbar from './SiteNavbar.vue';

/* Need to mock the `window.matchMedia` method here, because it has not
 * yet been implemented by JSDOM. Hopefully this will be fixed soon.
 * If we see errors creeping into other files about the `window.matchMedia`
 * method, then we can break this mock out into an includable file.
 *
 * https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
 */
Object.defineProperty(window, 'matchMedia', {
  value: jest.fn().mockImplementation((query) => ({
    addEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
    matches: false,
    media: query,
    onchange: null,
    removeEventListener: jest.fn(),
  })),
  writable: true,
});

describe('<SiteNavbar />', () => {
  it('should render without crashing', () => {
    const { unmount } = stubbedRender(SiteNavbar);

    unmount();
  });

  it('renders correctly', () => {
    const { container } = stubbedRender(SiteNavbar);

    expect(container.firstChild).toMatchSnapshot();
  });
});
