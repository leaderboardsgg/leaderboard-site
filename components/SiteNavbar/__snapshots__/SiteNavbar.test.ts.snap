// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SiteNavbar /> renders correctly 1`] = `
<div
  class="border-b border-gray-300"
>
  <div
    class="
      container
      relative
      flex flex-wrap
      lg:flex-row
      justify-between
      mx-auto
      px-3
      py-3
    "
  >
    <div
      class="flex flex-grow"
    >
      <a
        class="flex"
        href="/"
      >
        <svg
          class="h-10 w-10 mx-1"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
         
        <h1
          class="
            w-full
            py-2
            text-lg text-center
            md:text-left
            font-semibold
            leading-6
            tracking-tight
            hidden
            md:inline
          "
        >
          
          leaderboards.gg
        
        </h1>
      </a>
    </div>
     
    <div
      class="
        lg:flex
        border-t border-gray-300
        lg:border-t-0
        flex-col
        lg:flex-row
        mt-2
        lg:mt-0
        w-full
        lg:w-auto
        space-y-2
        lg:space-y-0
       hidden"
    >
      <nav
        class="nav-link-container"
      >
        <a
          class="button-link nav-link"
          href="#"
          name="Games"
          to="#"
        >
          
    Games
  
        </a>
        <a
          class="button-link nav-link"
          href="#"
          name="About"
          to="#"
        >
          
    About
  
        </a>
      </nav>
       
      <div
        class="search-bar"
      >
        <input
          data-testid="search-input"
          name="search"
          placeholder="search..."
        />
         
        <button
          data-testid="search-button"
          type="submit"
        >
          <svg
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
      </div>
    </div>
     
    <div
      class="
        flex flex-row
        items-center
        absolute
        right-2
        top-4
        lg:relative lg:right-0 lg:top-0
        space-x-2
        lg:space-x-0
      "
    >
      <button
        class="base-button login-button"
        data-testid="site-navbar-login-button"
      >
        
  Log In

      </button>
       
      <button
        class="base-button sign-up-button"
        data-testid="site-navbar-sign-up-button"
      >
        
  Sign Up

      </button>
       
      <button
        class="items-center rounded text-black lg:hidden"
      >
        <svg
          class="fill-current h-5 w-5 mx-2"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Menu
          </title>
           
          <path
            d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0v-2z"
          />
        </svg>
      </button>
    </div>
  </div>
   
  <transition-stub
    enteractiveclass="transition-opacity duration-200"
    enterclass="opacity-0"
    entertoclass="opacity-100"
    leaveactiveclass="transition-opacity duration-200"
    leaveclass="opacity-100"
    leavetoclass="opacity-0"
  >
    <div
      class="modal-wrapper"
      style="display: none;"
    >
      <div
        class="modal-container"
      >
        <div
          class="modal-fade"
        />
         
        <div
          class="modal-content"
        >
          <div
            class="card bg-white w-full max-w-xl shadow-xl"
            data-testid="login-card"
            id="loginCard"
            style="display: none;"
          >
            <div
              class="card-header flex flex-row space-x-3"
            >
              <div
                class="
        flex flex-1
        justify-center
        px-3
        py-2
        rounded
        bg-gray-100
        text-gray-900
      "
              >
                
      Log In
    
              </div>
               
              <button
                class="base-button 
        flex flex-1
        justify-center
        bg-white
        text-gray-900
        border border-gray-300
      "
                id="sign-up-button"
              >
                
      Sign Up
    
              </button>
               
              <button
                class="base-button close-button"
              >
                <svg
                  class="h-5 w-5"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
                    fill-rule="evenodd"
                  />
                </svg>
              </button>
            </div>
             
            <div
              class="card-body"
            >
              <div
                class="flex flex-col space-y-3 pb-3 mb-3 border-b border-gray-300"
              >
                <input
                  autocomplete="email"
                  class="input bg-white"
                  name="email"
                  placeholder="Email"
                  type="text"
                />
                 
                <div
                  class="relative flex flex-col w-full"
                >
                  <input
                    autocomplete="password"
                    class="input bg-white pr-12"
                    name="password"
                    placeholder="Password"
                    type="password"
                  />
                   
                  <div
                    class="absolute top-0 right-0 flex items-center h-full"
                  >
                    <button
                      class="base-button hidden sm:inline bg-white p-1 mr-2 rounded fill-current"
                      data-testid="hide-show-password-button"
                      id="hide-show-button"
                      type="button"
                    >
                      <svg
                        class="h-5 w-5"
                        data-testid="hidden-eye-icon"
                        fill="currentColor"
                        viewBox="0 0 20 20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M3.707 2.293a1 1 0 00-1.414 1.414l14 14a1 1 0 001.414-1.414l-1.473-1.473A10.014 10.014 0 0019.542 10C18.268 5.943 14.478 3 10 3a9.958 9.958 0 00-4.512 1.074l-1.78-1.781zm4.261 4.26l1.514 1.515a2.003 2.003 0 012.45 2.45l1.514 1.514a4 4 0 00-5.478-5.478z"
                          fill-rule="evenodd"
                        />
                         
                        <path
                          d="M12.454 16.697L9.75 13.992a4 4 0 01-3.742-3.741L2.335 6.578A9.98 9.98 0 00.458 10c1.274 4.057 5.065 7 9.542 7 .847 0 1.669-.105 2.454-.303z"
                        />
                      </svg>
                       
                      <svg
                        class="h-5 w-5"
                        data-testid="visible-eye-icon"
                        fill="currentColor"
                        style="display: none;"
                        viewBox="0 0 20 20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M10 12a2 2 0 100-4 2 2 0 000 4z"
                        />
                         
                        <path
                          clip-rule="evenodd"
                          d="M.458 10C1.732 5.943 5.522 3 10 3s8.268 2.943 9.542 7c-1.274 4.057-5.064 7-9.542 7S1.732 14.057.458 10zM14 10a4 4 0 11-8 0 4 4 0 018 0z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
                 
                <button
                  class="base-button login-button"
                  to="#"
                >
                  Log In
                </button>
              </div>
               
              <div
                class="flex flex-col w-full space-y-2"
              >
                <button
                  class="base-button login-button"
                >
                  <svg
                    class="w-5 h-5 mr-2"
                    viewBox="0 0 480 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M186.1 328.7c0 20.9-10.9 55.1-36.7 55.1s-36.7-34.2-36.7-55.1 10.9-55.1 36.7-55.1 36.7 34.2 36.7 55.1zM480 278.2c0 31.9-3.2 65.7-17.5 95-37.9 76.6-142.1 74.8-216.7 74.8-75.8 0-186.2 2.7-225.6-74.8-14.6-29-20.2-63.1-20.2-95 0-41.9 13.9-81.5 41.5-113.6-5.2-15.8-7.7-32.4-7.7-48.8 0-21.5 4.9-32.3 14.6-51.8 45.3 0 74.3 9 108.8 36 29-6.9 58.8-10 88.7-10 27 0 54.2 2.9 80.4 9.2 34-26.7 63-35.2 107.8-35.2 9.8 19.5 14.6 30.3 14.6 51.8 0 16.4-2.6 32.7-7.7 48.2 27.5 32.4 39 72.3 39 114.2zm-64.3 50.5c0-43.9-26.7-82.6-73.5-82.6-18.9 0-37 3.4-56 6-14.9 2.3-29.8 3.2-45.1 3.2-15.2 0-30.1-.9-45.1-3.2-18.7-2.6-37-6-56-6-46.8 0-73.5 38.7-73.5 82.6 0 87.8 80.4 101.3 150.4 101.3h48.2c70.3 0 150.6-13.4 150.6-101.3zm-82.6-55.1c-25.8 0-36.7 34.2-36.7 55.1s10.9 55.1 36.7 55.1 36.7-34.2 36.7-55.1-10.9-55.1-36.7-55.1z"
                      fill="currentColor"
                    />
                  </svg>
                   
                  <p>
                    Log In with Github
                  </p>
                </button>
                 
                <button
                  class="base-button login-button"
                >
                  <svg
                    class="w-5 h-5 mr-2"
                    viewBox="0 0 488 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z"
                      fill="currentColor"
                    />
                  </svg>
                   
                  <p>
                    Log In with Google
                  </p>
                </button>
              </div>
            </div>
          </div>
           
          <div
            class="card bg-white w-full max-w-xl shadow-xl"
            data-testid="sign-up-card"
            id="signUpCard"
          >
            <div
              class="card-header flex flex-row space-x-3"
            >
              <button
                class="base-button 
        flex flex-1
        justify-center
        bg-white
        text-gray-900
        border border-gray-300
      "
                id="login-button"
              >
                
      Log In
    
              </button>
               
              <div
                class="
        flex flex-1
        justify-center
        px-3
        py-2
        rounded
        bg-gray-100
        text-gray-900
      "
              >
                
      Sign Up
    
              </div>
               
              <button
                class="base-button close-button"
              >
                <svg
                  class="h-5 w-5"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
                    fill-rule="evenodd"
                  />
                </svg>
              </button>
            </div>
             
            <div
              class="p-3"
            >
              <div
                class="flex flex-col space-y-3 pb-3 mb-3 border-b border-gray-300"
              >
                <input
                  autocomplete="email"
                  class="input"
                  name="email"
                  placeholder="Email"
                  type="text"
                />
                 
                <div
                  class="flex flex-1 flex-col space-y-1"
                >
                  <input
                    autocomplete="nickname"
                    class="input"
                    name="username"
                    placeholder="Username"
                    type="text"
                  />
                   
                  <p
                    class="text-sm"
                  >
                    This can be changed later
                  </p>
                </div>
                 
                <div
                  class="flex flex-1 flex-col space-y-1"
                >
                  <div
                    class="
            flex flex-1 flex-col
            sm:flex-row
            space-y-3
            sm:space-y-0 sm:space-x-2
          "
                  >
                    <input
                      autocomplete="password"
                      class="input flex-grow"
                      name="password"
                      placeholder="Password"
                      type="password"
                    />
                     
                    <input
                      autocomplete="password"
                      class="input flex-grow"
                      name="confirmPassword"
                      placeholder="Confirm"
                      type="password"
                    />
                     
                    <button
                      class="base-button hidden sm:inline bg-white"
                      data-testid="hide-show-password-button"
                      id="hide-show-password"
                      type="button"
                    >
                      <svg
                        class="h-5 w-5"
                        data-testid="hidden-eye-icon"
                        fill="currentColor"
                        viewBox="0 0 20 20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M3.707 2.293a1 1 0 00-1.414 1.414l14 14a1 1 0 001.414-1.414l-1.473-1.473A10.014 10.014 0 0019.542 10C18.268 5.943 14.478 3 10 3a9.958 9.958 0 00-4.512 1.074l-1.78-1.781zm4.261 4.26l1.514 1.515a2.003 2.003 0 012.45 2.45l1.514 1.514a4 4 0 00-5.478-5.478z"
                          fill-rule="evenodd"
                        />
                         
                        <path
                          d="M12.454 16.697L9.75 13.992a4 4 0 01-3.742-3.741L2.335 6.578A9.98 9.98 0 00.458 10c1.274 4.057 5.065 7 9.542 7 .847 0 1.669-.105 2.454-.303z"
                        />
                      </svg>
                       
                      <svg
                        class="h-5 w-5"
                        data-testid="visible-eye-icon"
                        fill="currentColor"
                        style="display: none;"
                        viewBox="0 0 20 20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M10 12a2 2 0 100-4 2 2 0 000 4z"
                        />
                         
                        <path
                          clip-rule="evenodd"
                          d="M.458 10C1.732 5.943 5.522 3 10 3s8.268 2.943 9.542 7c-1.274 4.057-5.064 7-9.542 7S1.732 14.057.458 10zM14 10a4 4 0 11-8 0 4 4 0 018 0z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </button>
                  </div>
                   
                  <p
                    class="text-sm"
                  >
                    
          * Must be 8 characters with a mix of letters and numbers
        
                  </p>
                </div>
                 
                <button
                  class="base-button signup-button"
                >
                   Sign Up 
                </button>
              </div>
               
              <div
                class="flex flex-col w-full space-y-2"
              >
                <button
                  class="base-button signup-button"
                >
                  <svg
                    class="w-5 h-5 mr-2"
                    viewBox="0 0 480 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M186.1 328.7c0 20.9-10.9 55.1-36.7 55.1s-36.7-34.2-36.7-55.1 10.9-55.1 36.7-55.1 36.7 34.2 36.7 55.1zM480 278.2c0 31.9-3.2 65.7-17.5 95-37.9 76.6-142.1 74.8-216.7 74.8-75.8 0-186.2 2.7-225.6-74.8-14.6-29-20.2-63.1-20.2-95 0-41.9 13.9-81.5 41.5-113.6-5.2-15.8-7.7-32.4-7.7-48.8 0-21.5 4.9-32.3 14.6-51.8 45.3 0 74.3 9 108.8 36 29-6.9 58.8-10 88.7-10 27 0 54.2 2.9 80.4 9.2 34-26.7 63-35.2 107.8-35.2 9.8 19.5 14.6 30.3 14.6 51.8 0 16.4-2.6 32.7-7.7 48.2 27.5 32.4 39 72.3 39 114.2zm-64.3 50.5c0-43.9-26.7-82.6-73.5-82.6-18.9 0-37 3.4-56 6-14.9 2.3-29.8 3.2-45.1 3.2-15.2 0-30.1-.9-45.1-3.2-18.7-2.6-37-6-56-6-46.8 0-73.5 38.7-73.5 82.6 0 87.8 80.4 101.3 150.4 101.3h48.2c70.3 0 150.6-13.4 150.6-101.3zm-82.6-55.1c-25.8 0-36.7 34.2-36.7 55.1s10.9 55.1 36.7 55.1 36.7-34.2 36.7-55.1-10.9-55.1-36.7-55.1z"
                      fill="currentColor"
                    />
                  </svg>
                   
                  <p>
                    Sign Up with Github
                  </p>
                </button>
                 
                <button
                  class="base-button signup-button"
                >
                  <svg
                    class="w-5 h-5 mr-2"
                    viewBox="0 0 488 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z"
                      fill="currentColor"
                    />
                  </svg>
                   
                  <p>
                    Sign Up with Google
                  </p>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </transition-stub>
</div>
`;
